packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateFormV2.tsx

```diff
@@ -0,0 +1,308 @@
+        $name={plan.product_name}
+        $date={getLocalizedDate(customerSubscription.current_period_end, true)}
+      >
+        <p className="billing-description">{billingDescriptionText}</p>
+      </Localized>
+      {upgradeCTA && (
+        <p
+          className="upgrade-cta"
+          data-testid="upgrade-cta"
+          dangerouslySetInnerHTML={{ __html: upgradeCTA }}
+        />
+      )}
+      {!updateRevealed ? (
+        <div className="with-settings-button">
+          <div className="card-details">
+            {last4 && expirationDate && (
+              <>
+                {/* TODO: Need to find a way to display a card icon here? */}
+                <Localized id="sub-update-card-ending" $last={last4}>
+                  <div className="last-four">Card ending {last4}</div>
+                </Localized>
+                <Localized
+                  id="pay-update-card-exp"
+                  $expirationDate={expirationDate}
+                >
+                  <div data-testid="card-expiration-date" className="expiry">
+                    Expires {expirationDate}
+                  </div>
+                </Localized>
+              </>
+            )}
+          </div>
+          <div className="action">
+            <button
+              data-testid="reveal-payment-update-button"
+              className="settings-button"
+              onClick={onRevealUpdateClick}
+            >
+              <Localized id="pay-update-change-btn">
+                <span className="change-button" data-testid="change-button">
+                  Change
+                </span>
+              </Localized>
+            </button>
+          </div>
+        </div>
+      ) : (
+        <>
+          <ErrorMessage isVisible={!!paymentError}>
+            {paymentError && (
+              <Localized id={getErrorMessage(paymentError.code || 'UNKNOWN')}>
+                <p data-testid="error-payment-submission">
+                  {getErrorMessage(paymentError.code || 'UNKNOWN')}
+                </p>
+              </Localized>
+            )}
+          </ErrorMessage>
+          <PaymentForm
+            {...{
+              submitNonce,
+              onSubmit,
+              inProgress,
+              confirm: false,
+              onCancel: hideUpdate,
+              onChange,
+              onMounted: onFormMounted,
+              onEngaged: onFormEngaged,
+            }}
+          />
+        </>
+      )}
+    </div>
+  );
+};
+
+function getBillingDescriptionText(
+  name: string,
+  amount: number,
+  currency: string,
+  interval: PlanInterval,
+  intervalCount: number,
+  date: number
+): string {
+  const formattedDateString = getLocalizedDateString(date, true);
+  const planPricing = formatPlanPricing(
+    amount,
+    currency,
+    interval,
+    intervalCount
+  );
+
+  return `You are billed ${planPricing} for ${name}. Your next payment occurs on ${formattedDateString}.`;
+}
+
+async function handlePaymentUpdate({
```

Comment:
Oh yeah, and this `handlePaymentUpdate` function is the core of the new billing update flow.